stone/frontend/ir_generator.py

```diff
@@ -172,6 +181,8 @@ def generate_IR(self):
         self._populate_examples()
         self._validate_doc_refs()
         self._validate_annotations()
+        if self._routes is not None:
```

Comment:
this can be simplified to `if self._routes`.  The additional `is not None` is redundant.